plugin/output/s3/s3.go

```diff
@@ -234,22 +258,26 @@ func (p *Plugin) uploadWork() {
 
 // compressWork compress file from channel and then delete source file
 func (p *Plugin) compressWork() {
-	for f := range p.compressCh {
-		compressedName := p.compressor.getName(f)
-		p.compressor.compress(compressedName, f)
+	for dto := range p.compressCh {
+		compressedName := p.compressor.getName(dto.fileName)
+		p.compressor.compress(compressedName, dto.fileName)
 		// delete old file
-		if err := os.Remove(f); err != nil {
-			p.logger.Panicf("could not delete file: %s, error: %s", f, err.Error())
+		if err := os.Remove(dto.fileName); err != nil {
+			p.logger.Panicf("could not delete file: %s, error: %s", dto, err.Error())
 		}
-		p.uploadCh <- compressedName
+		dto.fileName = compressedName
+		p.uploadCh <- fileDto{fileName: dto.fileName, bucketName: dto.bucketName}
 	}
 }
 
-// uploadToS3 uploads compressed file to s3
-func (p *Plugin) uploadToS3(name string) error {
-	_, err := p.client.FPutObject(p.config.Bucket, p.generateObjectName(name), name, p.compressor.getObjectOptions())
+func (p *Plugin) uploadToS3(compressedDto fileDto) error {
+	_, err := p.clients[compressedDto.bucketName].FPutObject(
+		p.config.Bucket, p.generateObjectName(compressedDto.fileName),
```

Comment:
Here we get client by compressedDto.bucketName, but call FPutObject with default bucket. Need to change it.

Reply:
Thx, that was a bug. I'll take up minio server probably for E2E testing.